extensions/resource-deployment/src/ui/deployAzureSQLDBWizard/deployAzureSQLDBWizard.ts

```diff
@@ -172,4 +189,33 @@ export class DeployAzureSQLDBWizard extends WizardBase<DeployAzureSQLDBWizard, W
 			level: azdata.window.MessageLevel.Error
 		};
 	}
+
+	public async refreshPages() {
+
+		const pageCount = this.wizardObject.pages.length;
+		for (let i = 1; i < pageCount; i++) {
+			this.wizardObject.removePage(this.wizardObject.pages.length - 1);
+			this.wizardObject.pages.pop();
+		}
+		if (instanceOfAzureSQLDBDeploymentProvider(this.resourceProvider)) {
```

Comment:
How about for all thse wizards that have a different wizard for a different provider we create a common base class. In the refreshPages of that common class you can write a switch statement that will switch on different types of resourceProvider to set the right wizardInfo object then the right thing will happen as each subclass calls super.refreshPages() up the chain.